Shortcodes: minify jQuery Cycle script

eliorivero opened the pull request:
I noticed this on my new site elio.blog and searched for an issue here and there's this one that this PR now fixes #3335. I know that #5620 exists but I'm not sure if we need to minify these kind of files on each build since we won't be changing them often. So, here's this PR that specifically solves #3335.

Also, the reason why I noticed this script was because after running a test in the site I saw the script loaded, and I wasn't using it at all. I realized that when Infinite Scroll is supported, this script and others are always loaded. However, in my tests this wasn't true. I tested with this same branch and whenever the post that had a slideshow gallery was loaded, the script was loaded, so it's not necessary to always load it.

#### Changes proposed in this Pull Request:
- use minified jQuery Cycle JS
- don't load the script always when Infinite Scroll is activated

#### Testing instructions:
* add a slideshow gallery and verify it works normally
* test Infinite Scroll and add a slideshow in a post that is displayed after IS fetches more entries and verify it works

<!-- Add the following only if this is meant to be in changelog -->
#### Proposed changelog entry for your changes:
Shortcodes: the jQuery Cycle script used by slideshow galleries is now minified resulting in faster loading times.

cc @lancewillett for review

lancewillett commented on modules/shortcodes/slideshow.php:
```diff
@@ -273,23 +265,9 @@ function slideshow_js( $attr ) {
-			return;
-		}
 
 		wp_enqueue_script( 'jquery-cycle', plugins_url( '/js/jquery.cycle.js', __FILE__ ), array( 'jquery' ), '2.9999.8', true );
```
Do we need to bump the version number here to bust cache?

lancewillett commented:
Tested locally with Jetpack dev environment, works well. Going to give it a spin on WP.com test site also.

lancewillett commented:
Tested successfully on WP.com. 👍  (I renamed the file there to add in `min`, though.)

eliorivero commented:
@lancewillett I have:
- renamed file to jquery.cycle.min
- updated version to bust cache
- re-enabled the tests and modify the slideshow.php file to pass tests

lancewillett commented:
@eliorivero Tests out both Jetpack local and on WP.com, looks great.

dereksmart commented:
Nice, LGTM -- Thanks @lancewillett for testing on wpcom!

dereksmart commented:
merged to 4.5 fb7e160
